src/dmd/mars.d

```diff
@@ -904,6 +904,23 @@ version (NoMain) {} else
     {
         import core.runtime;
         import core.memory;
+        // install the memory error handler when the environment variable is set
```

Comment:
```
src/dmd/mars.d:908:16: error: module memoryerror is in file 'etc/linux/memoryerror.d' which cannot be read
  908 |         import etc.linux.memoryerror;
      |                ^
import path[0] = /usr/lib/gcc/x86_64-linux-gnu/9/include/d
import path[1] = /home/runner/dmd/src
```

As LDC or GDC might be host compilers too, I would simply use `version (DigitalMars)`.

Reply:
If we want to merge this to test further, I will. For now, just trying to see if I can duplicate the problem on this PR.

Reply:
- AFAICT the problem is a (fixed) segfault in GC (i.e. appears only with `-lowmem` + only old compilers)?
- this should fix the problem https://github.com/dlang/dmd/pull/11519

I still believe there's merit of keeping the memory error registration enabled as it will likely help with other hard-to-replicate bugs that appear on the CI machines.